ipldutils/ipldutils.go

```diff
@@ -0,0 +1,169 @@
+package shared
+
```

Comment:
What is going on with all this code at this point? It seems like a bunch of complexity maybe designed to support actually picking bindnode schemas when deserializing data transfer vouchers, but it's only used for the actual data transfer message formats. Do we need it? does this actually make the message code easier to read?

Reply:
TODO - it's copypasta from go-fil-markets code currently which is evolving to deal with message complexity there. But it's probably way too much here.

My thought was to extract this to a separate package but let's discuss this. The idea was that we could hide the schemas within the type itself so you just need to pass around type pointers and you don't need to pre-register for prototypes, it's done lazily and cached...

Mostly I'd really like to be using similar patterns across both codebases for this bindnode work, somehow, if possible.